src/librustc_trans/context.rs

```diff
@@ -322,6 +322,7 @@ unsafe fn create_context_and_module(sess: &Session, mod_name: &str) -> (ContextR
     let llvm_target = sess.target.target.llvm_target.as_bytes();
     let llvm_target = CString::new(llvm_target).unwrap();
     llvm::LLVMRustSetNormalizedTarget(llmod, llvm_target.as_ptr());
+    llvm::LLVMRustSetModulePIELevel(llmod);
```

Comment:
This is actually currently conditional when the argument is passed to `LLVMRustCreateTargetMachine`:

```rust
!any_library && reloc_model == llvm::RelocPIC
```

Perhaps that logic could be extracted to a method and that could be passed down here as well?